### PR TITLE
S1-ONE-120 Sign up form

### PR DESCRIPTION
**Pull request by evasquez26**
## Story / Task

[ONE-120](https://bixlabs.atlassian.net/browse/ONE-120)

## What this PR does?

Allows the user to register into the app.

### Before 

We could not register into the application

### After

User can register into the application using his email address

### How to test / reproduce

- Run the app without being logged in
- Observe that there's a button at the top right that says 'Sign up"
- Clicking the button should redirect you to the sign up form
- All fields are required, if the user submits the form without some fields it will mark the fields that are blank with error
- Filling all the fields should let you advance to the sign up success form

### TODOs
- [x] Correct implementation with the real API
- [ ] Test the confirmation process

### Related PRs

N/A

## Migrations

NO

Thanks!

### CODE REVIEW

File: src/app/signup/signup.component.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            <p *ngIf="!lastNameField.valid && submitted" class="text-warning error-message">Last Name is required</p>
+          </div>
+          <div [ngClass]="{'has-error': !email.valid && submitted}" class="form-group">
+            <input [(ngModel)]="registerRequest.email" name="email" #email="ngModel" id="email" class="form-control" required type="email" placeholder="Email">
+            <p *ngIf="!email.valid && submitted" class="text-warning error-message">{{errorEmail}}</p>
+          </div>
+          <div [ngClass]="{'has-error': !password.valid && submitted}" class="form-group">
+            <input [(ngModel)]="registerRequest.password" name="password" #password="ngModel" id="password" class="form-control" minlength="6" required type="password" placeholder="Password">
+            <p *ngIf="!password.valid && !passwordError && submitted" class="text-warning error-message">Password is required</p>
+            <p *ngIf="!password.valid && passwordError && submitted" class="text-warning error-message">Password must be 6 characters min long</p>
```
Comment (iwannabaa) [inline src/app/signup/signup.component.html]: It actually must be longer than 6 characters

File: src/app/signup/signup.component.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            <input [(ngModel)]="firstName" name="first-name" #firstNameField="ngModel" id="firstName" class="form-control" required type="text" placeholder="Name">
+            <p *ngIf="!firstNameField.valid && submitted" class="text-warning error-message">First Name is required</p>
+          </div>
+          <div [ngClass]="{'has-error': !lastNameField.valid && submitted}" class="form-group">
+            <input [(ngModel)]="lastName" name="last-name" #lastNameField="ngModel" id="lastName" class="form-control" required type="text" placeholder="Last Name">
+            <p *ngIf="!lastNameField.valid && submitted" class="text-warning error-message">Last Name is required</p>
+          </div>
+          <div [ngClass]="{'has-error': !email.valid && submitted}" class="form-group">
+            <input [(ngModel)]="registerRequest.email" name="email" #email="ngModel" id="email" class="form-control" required type="email" placeholder="Email">
+            <p *ngIf="!email.valid && submitted" class="text-warning error-message">{{errorEmail}}</p>
```
Comment (iwannabaa) [inline src/app/signup/signup.component.html]: If I write an invalid email (i.e. without  `@domain.com`) it does not show error

File: src/app/shared/utils.service.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class UtilsService {
```
Comment (iwannabaa) [inline src/app/shared/utils.service.ts]: Perfect! Thanks for moving to a service

File: src/app/signup/shared/signup.service.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  register(registerRequest: RegisterRequest) {
+    const url = `${CONFIG.api.url}/core/users`;
+
+    const headers = new Headers({ 'Content-Type': 'application/json' });
+    const options = new RequestOptions({ headers: headers });
+
+    return this.http
+               .post(url, registerRequest)
+               .map(response => <AuthResponse>response.json())
+               .catch(error => Observable.throw(error));
```
Comment (iwannabaa) [inline src/app/signup/shared/signup.service.ts]: Use the exception service here and in the confirm function

File: src/app/shared/utils.service.ts (hunk 3 above)
Comment (evasquez26) [inline src/app/shared/utils.service.ts]: I just realized that both email and password validations were unnecessary as the angular directives `minlength` and `email` take care of that

**Pull request merged**